lib/v2/screens/explore_screens/invite/interactor/mappers/create_invite_result_state_mapper.dart

```diff
@@ -0,0 +1,25 @@
+// import 'package:seeds/v2/datasource/remote/model/transaction_response.dart';
+import 'package:seeds/v2/domain-shared/page_state.dart';
+import 'package:seeds/v2/domain-shared/result_to_state_mapper.dart';
+import 'package:seeds/v2/screens/explore_screens/invite/interactor/viewmodels/invite_state.dart';
+
+class CreateInviteResultStateMapper extends StateMapper {
+  InviteState mapResultsToState(InviteState currentState, List<Result> results) {
+    if (areAllResultsError(results)) {
+      return currentState.copyWith(pageState: PageState.failure, errorMessage: 'Error creating invite');
+    } else {
+      print('CreateInviteResultStateMapper mapResultsToState length = ${results.length}');
+      results.retainWhere((Result element) => element.isValue);
+      var values = results.map((Result element) => element.asValue!.value).toList();
+
+      // TransactionResponse? response = values.firstWhere((i) => i is TransactionResponse, orElse: () => null);
+      Uri? dynamicSecretLink = values.firstWhere((i) => i is Uri, orElse: () => null);
+
+      return currentState.copyWith(
+        pageState: PageState.success,
+        showInviteLinkDialog: true,
+        dynamicSecretLink: dynamicSecretLink.toString(),
+      );
```

Comment:
I'm not sure what to do if the transaction fails but the firebase dynamic link is met. Should it show an error page if the transaction to create the invitation fails?

```
if(response.transactionId.isEmpty){
// error page
}else{
// success flow
}
```

Reply:
yeah, probably show and error and ask the user to retry?